seat.c

```diff
@@ -607,20 +608,25 @@ handle_cursor_button(struct wl_listener *listener, void *data)
 }
 
 static void
-process_cursor_motion(struct cg_seat *seat, uint32_t time)
+process_cursor_motion(struct cg_seat *seat, uint32_t time_msec, double dx, double dy, double dx_unaccel,
+		      double dy_unaccel)
 {
 	double sx, sy;
 	struct wlr_seat *wlr_seat = seat->seat;
 	struct wlr_surface *surface = NULL;
 
-	struct cg_view *view = desktop_view_at(seat->server, seat->cursor->x, seat->cursor->y, &surface, &sx, &sy);
+	if (dx != 0 || dy != 0) {
+		wlr_relative_pointer_manager_v1_send_relative_motion(seat->server->relative_pointer_manager, seat->seat,
```

Comment:
Hm, this looks a bit weird. It feels like this would always send the cursor event to the _previous_ surface, even if the event moves the cursor over another surface?

Can we move this right next to `wlr_seat_pointer_notify_motion()` instead?

Reply:
You are right, I was missing this point. Just moved this part of code.